File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode2248.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function intersection(nums) {
    var n = nums.length;
    var map = new Map();
    for (var _i = 0, nums_1 = nums; _i < nums_1.length; _i++) {
        var arr = nums_1[_i];
        for (var _a = 0, arr_1 = arr; _a < arr_1.length; _a++) {
            var i = arr_1[_a];
            map.set(i, (map.get(i) || 0) + 1);
        }
    }
    var result = [];
    for (var _b = 0, _c = map.entries(); _b < _c.length; _b++) {
        var _d = _c[_b], key = _d[0], value = _d[1];
        if (value === n) {
            result.push(key);
        }
    }
    result.sort(function (a, b) { return a + b; });
    return result;
}
var nums = [[3, 1, 2, 4, 5], [1, 2, 3, 4], [3, 4, 5, 6]];
console.log(intersection(nums));

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode2248.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

public class Leetcode2248 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the row number: ");
        int row = input.nextInt();
        System.out.print("Enter the column number: ");
        int col = input.nextInt();
        int[][] arr = new int[row][col];

        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                System.out.print("Enter the element of row: " + (i + 1) + " column: " + (j + 1) + ": ");
                arr[i][j] = input.nextInt();
            }
        }
        input.close();
        System.out.println(intersection(arr));
    }

    private static List<Integer> intersection(int[][] nums) {
        int n = nums.length;
        Map<Integer, Integer> map = new HashMap<>();
        for (int[] arr : nums) {
            for (int i : arr) {
                map.put(i, map.getOrDefault(i, 0) + 1);
            }
        }
        List<Integer> result = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == n) {
                result.add(entry.getKey());
            }
        }
        Collections.sort(result);
        return result;
    }
}